parser/src/lexer.rs

```diff
@@ -1397,6 +1461,183 @@ mod tests {
         }
     }
 
+    macro_rules! test_jupyter_magic_line_continuation_eol {
+        ($($name:ident: $eol:expr,)*) => {
+            $(
+            #[test]
+            fn $name() {
+                let source = format!("%matplotlib \\{}  --inline", $eol);
+                let tokens = lex_jupyter_source(&source);
+                assert_eq!(
+                    tokens,
+                    vec![
+                        Tok::MagicCommand { value: "matplotlib   --inline".to_string(), kind: MagicKind::Magic },
+                    ]
+                )
+            }
+            )*
+        };
+    }
```

Comment:
Nit: I personally find macros hard to understand. I think we can achieve the same with slightly more boilerplate

```rust
def assert_jupyter_magic_line_continuation_with_eol(eol &str) {
	let source = format!("%matplotlib \\{}  --inline", eol);
    let tokens = lex_jupyter_source(&source);
    assert_eq!(
        tokens,
        vec![
            Tok::MagicCommand { value: "matplotlib   --inline".to_string(), kind: MagicKind::Magic },
        ]
    )
}

#[test]
def fn test_jupyter_magic_continuation_windows_eol() {
	assert_jupyter_magic_line_continuation_with_eol(WINDOWS_EOL);
}


#[test]
def fn test_jupyter_magic_continuation_mac_eol() {
	assert_jupyter_magic_line_continuation_with_eol(MAC_EOL);
}

...
```